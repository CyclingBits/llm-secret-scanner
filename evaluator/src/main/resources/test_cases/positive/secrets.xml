<?xml version="1.0" encoding="UTF-8"?>
<!-- deploy-token-from-xml-comment-xyz123 -->
<configuration>
    <properties>
        <property name="database.password" value="xml_database_password_2024!" />
        <property name="api.key" value="xml_api_key_abc123def456" />
        <property name="secret.token" value="xml_secret_token_999" />
        <property name="encoded.data" value="eG1sLWJhc2U2NC1zZWNyZXQtZGF0YQ==" />
    </properties>

    <datasources>
        <datasource>
            <name>production</name>
            <url>jdbc:postgresql://user:xml_db_password@db.example.com:5432/prod_db</url>
            <username>admin</username>
            <password>xml_datasource_password_456</password>
        </datasource>
    </datasources>

    <certificates>
        <certificate type="private-key">
            <data>
-----BEGIN RSA PRIVATE KEY-----
MIIEpAIBAAKCAQEAxml_private_key_data_here_for_testing_only
VGVzdCBwcml2YXRlIGtleSBmb3IgWE1MIHRFU3RpbmcgLSBub3QgcmVhbA==
-----END RSA PRIVATE KEY-----
            </data>
        </certificate>
    </certificates>

    <authentication>
        <oauth>
            <client-id>xml_oauth_client_123</client-id>
            <client-secret>xml_oauth_secret_789</client-secret>
            <token-endpoint>https://api.example.com/oauth/token</token-endpoint>
        </oauth>
        
        <!-- session_key_xml_commented_abc -->
        
        <jwt>
            <secret-key>xml_jwt_secret_key_def456</secret-key>
            <issuer>xml-service</issuer>
            <expiry>3600</expiry>
        </jwt>
    </authentication>

    <security>
        <encryption>
            <master-key>xml_master_encryption_key_ghi789</master-key>
            <salt>xml_encryption_salt_jkl012</salt>
        </encryption>
        
        <api-keys>
            <key name="dev">xml_dev_api_key_111</key>
            <key name="staging">xml_staging_api_key_222</key>
            <key name="prod">xml_prod_api_key_333</key>
        </api-keys>
    </security>

    <external-services>
        <service name="payment-gateway">
            <endpoint>https://payments.example.com/api</endpoint>
            <auth-token>xml_payment_auth_token_mno345</auth-token>
            <api-key>xml_payment_api_key_pqr678</api-key>
        </service>
        
        <service name="email-service">
            <smtp-password>xml_smtp_password_stu901</smtp-password>
            <api-secret>xml_email_api_secret_vwx234</api-secret>
        </service>
    </external-services>

    <monitoring>
        <metrics>
            <api-key>xml_metrics_api_key_yz567</api-key>
        </metrics>
        
        <logging>
            <secret-key>xml_logging_secret_key_890</secret-key>
        </logging>
    </monitoring>

    <defaults>
        <admin-password>xml_default_admin_password_abc123</admin-password>
        <system-token>xml_system_token_def456</system-token>
    </defaults>

    <cdata-secrets>
        <private-certificate><![CDATA[
-----BEGIN CERTIFICATE-----
xml_cdata_certificate_secret_ghi789
VGVzdCBjZXJ0aWZpY2F0ZSBkYXRhIGZvciBYTUwgQ0RBVEEgc2VjdGlvbg==
-----END CERTIFICATE-----
        ]]></private-certificate>
        
        <connection-string><![CDATA[mongodb://admin:xml_cdata_mongo_pass@cluster.example.com:27017/prod]]></connection-string>
    </cdata-secrets>

    <nested>
        <deeply>
            <nested>
                <secret>xml_deeply_nested_secret_jkl012</secret>
            </nested>
        </deeply>
    </nested>

    <!-- Multi-line comment with secret:
         xml_multiline_comment_secret_mno345
         This should be detected
    -->

    <conditional-secrets>
        <secret condition="dev">xml_conditional_dev_secret_pqr678</secret>
        <secret condition="prod">xml_conditional_prod_secret_stu901</secret>
    </conditional-secrets>

</configuration>