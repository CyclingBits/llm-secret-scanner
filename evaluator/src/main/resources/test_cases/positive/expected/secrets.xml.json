[
  {
    "filePath": "secrets.xml",
    "issueNumber": 1,
    "lineNumber": 2,
    "secretValue": "deploy-token-from-xml-comment-xyz123"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 2,
    "lineNumber": 5,
    "secretValue": "xml_database_password_2024!"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 3,
    "lineNumber": 6,
    "secretValue": "xml_api_key_abc123def456"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 4,
    "lineNumber": 7,
    "secretValue": "xml_secret_token_999"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 5,
    "lineNumber": 8,
    "secretValue": "eG1sLWJhc2U2NC1zZWNyZXQtZGF0YQ=="
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 6,
    "lineNumber": 14,
    "secretValue": "xml_db_password"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 7,
    "lineNumber": 16,
    "secretValue": "xml_datasource_password_456"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 8,
    "lineNumber": 23,
    "secretValue": "-----BEGIN RSA PRIVATE KEY-----"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 9,
    "lineNumber": 34,
    "secretValue": "xml_oauth_secret_789"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 10,
    "lineNumber": 38,
    "secretValue": "session_key_xml_commented_abc"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 11,
    "lineNumber": 41,
    "secretValue": "xml_jwt_secret_key_def456"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 12,
    "lineNumber": 49,
    "secretValue": "xml_master_encryption_key_ghi789"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 13,
    "lineNumber": 50,
    "secretValue": "xml_encryption_salt_jkl012"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 14,
    "lineNumber": 54,
    "secretValue": "xml_dev_api_key_111"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 15,
    "lineNumber": 55,
    "secretValue": "xml_staging_api_key_222"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 16,
    "lineNumber": 56,
    "secretValue": "xml_prod_api_key_333"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 17,
    "lineNumber": 63,
    "secretValue": "xml_payment_auth_token_mno345"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 18,
    "lineNumber": 64,
    "secretValue": "xml_payment_api_key_pqr678"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 19,
    "lineNumber": 68,
    "secretValue": "xml_smtp_password_stu901"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 20,
    "lineNumber": 69,
    "secretValue": "xml_email_api_secret_vwx234"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 21,
    "lineNumber": 75,
    "secretValue": "xml_metrics_api_key_yz567"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 22,
    "lineNumber": 79,
    "secretValue": "xml_logging_secret_key_890"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 23,
    "lineNumber": 84,
    "secretValue": "xml_default_admin_password_abc123"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 24,
    "lineNumber": 85,
    "secretValue": "xml_system_token_def456"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 25,
    "lineNumber": 90,
    "secretValue": "-----BEGIN CERTIFICATE-----"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 26,
    "lineNumber": 96,
    "secretValue": "xml_cdata_mongo_pass"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 27,
    "lineNumber": 102,
    "secretValue": "xml_deeply_nested_secret_jkl012"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 28,
    "lineNumber": 108,
    "secretValue": "xml_multiline_comment_secret_mno345"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 29,
    "lineNumber": 113,
    "secretValue": "xml_conditional_dev_secret_pqr678"
  },
  {
    "filePath": "secrets.xml",
    "issueNumber": 30,
    "lineNumber": 114,
    "secretValue": "xml_conditional_prod_secret_stu901"
  }
]
